internal/app/admin/http/cache.go

```diff
@@ -210,6 +228,17 @@ func printCacheEntries(keys []string, cache cache.ReadOnlyCache, w http.Response
 	}
 }
 
+func clearCacheEntries(keys []string, cache cache.Cache, o *orchestrator.Orchestrator, w http.ResponseWriter) {
+	for _, key := range keys {
+		resource, err := cache.DeleteKey(key)
```

Comment:
is there necessity for doing this operation inside a lock?

Reply:
would it be better if this operation was part of orchestrator?

Reply:
I moved it to the orchestrator, but I don't believe a lock is necessary. Key and stream deletion all happen under the same cache lock, so the cache and streams are protected from being out of sync. A lock around the entire operation would prevent multiple ClearCache handler calls from running at the same time, but that is not necessary. The DeleteKey() operation is safe against concurrent removals of the same key (we both acquire a lock on the cache before attempting to delete and within first check if the key still exists within the lock).